src/simulation/dynamics/_GeneralModuleFiles/svIntegratorRungeKutta.h

```diff
@@ -0,0 +1,208 @@
+                                           const ExtendedStateVector& states);
+
+    /**
+     * Computes the "k" coefficients of the Runge-Kutta method
+     * for a time and state.
+     */
+    KCoefficientsValues computeKCoefficients(double currentTime, double timeStep,
+                                             const ExtendedStateVector& currentStates);
+
+    /**
+     * Adds the "k" coefficients, weighted by the "c" coefficients
+     * to find the state after the time step.
+     */
+    ExtendedStateVector computeNextState(double timeStep, const ExtendedStateVector& currentStates,
+                                         const KCoefficientsValues& kVectors);
+
+  protected:
+    // coefficients is stored as a pointer to support polymorphism
+    /** Coefficients to be used in the method */
+    const std::unique_ptr<RKCoefficients<numberStages>> coefficients;
+};
+
+template <size_t numberStages>
+svIntegratorRungeKutta<numberStages>::svIntegratorRungeKutta(
+    DynamicObject* dynIn, const RKCoefficients<numberStages>& coefficients)
+    : StateVecIntegrator(dynIn),
+      coefficients(std::make_unique<RKCoefficients<numberStages>>(coefficients))
+{
+}
+
+template <size_t numberStages>
+svIntegratorRungeKutta<numberStages>::svIntegratorRungeKutta(
+    DynamicObject* dynIn, std::unique_ptr<RKCoefficients<numberStages>>&& coefficients)
+    : StateVecIntegrator(dynIn), coefficients(std::move(coefficients))
+{
+}
+
+template <size_t numberStages>
+void svIntegratorRungeKutta<numberStages>::integrate(double currentTime, double timeStep)
+{
+    ExtendedStateVector currentState = ExtendedStateVector::fromStates(this->dynPtrs);
+    KCoefficientsValues kValues = this->computeKCoefficients(currentTime, timeStep, currentState);
+    ExtendedStateVector nextState = this->computeNextState(timeStep, currentState, kValues);
+    nextState.setStates(this->dynPtrs);
+}
+
+template <size_t numberStages>
+ExtendedStateVector
+svIntegratorRungeKutta<numberStages>::computeDerivatives(double time, double timeStep,
+                                                         const ExtendedStateVector& states)
+{
+    states.setStates(this->dynPtrs);
+
+    for (auto dynPtr : this->dynPtrs) {
+        dynPtr->equationsOfMotion(time, timeStep);
+    }
+
+    return ExtendedStateVector::fromStateDerivs(this->dynPtrs);
+}
+
+template <size_t numberStages>
+auto svIntegratorRungeKutta<numberStages>::computeKCoefficients(
+    double currentTime, double timeStep, const ExtendedStateVector& currentStates)
+    -> KCoefficientsValues
+{
+    KCoefficientsValues kVectors;
+
+    for (size_t stageIndex = 0; stageIndex < numberStages; stageIndex++) {
+        double timeToComputeK = currentTime + this->coefficients->cArray.at(stageIndex) * timeStep;
+
+        if (stageIndex == 0) // Avoids one ExtendedStateVector copy
+        {
+            kVectors.at(stageIndex) =
+                this->computeDerivatives(timeToComputeK, timeStep, currentStates);
```

Comment:
The clang-formatter did that automatically to avoid going over the 100 columns, which is my personal max for C++. I guess the clang format prefers that over:
```
kVectors.at(stageIndex) = this->computeDerivatives(
    timeToComputeK, 
    timeStep, 
    currentStates
);
```
I can increase the column limit or just write the above format manually, if prefered. 

Btw, it would also great to have `.clang-format` file for the repository to prevent these issues.

Reply:
Agreed.